openwisp-monitoring/files/monitoring.agent

```diff
@@ -276,11 +278,29 @@ main() {
 	INTERVAL=${INTERVAL:-300}
 	REGISTRATION_INTERVAL=$((INTERVAL / 10))
 	VERBOSE_MODE=${VERBOSE_MODE:-0}
+	BOOTUP_DELAY=${BOOTUP_DELAY:-10}
 	TMP_DIR="/tmp/openwisp/monitoring"
+	BOOTUP="$TMP_DIR/bootup"
 
 	[ -z "$MODE" ] && echoerr "missing required --mode option"
 
+	# make directory
+	mkdir -p "$TMP_DIR"
+
 	if [ "$MODE" = "collect" ]; then
+		if [ ! -f "$BOOTUP" ]; then
```

Comment:
I think moving this block of code in its own function and calling only the function here (eg `bootup_delay`) would improve readability and maintain consistency with the rest of the code in this file.